### ISSUE TITLE
[Feature Request] Add prefix name for the title 

### ISSUE THREAD

**Issue by hany606** [Future-Scholars/paperlib#530]
Add an extra field in the metadata for a prefix name of the document. This prefix name is assigned by the user. Sometimes, I want to add the method name (which is not simply the {firstchartitle})

So, I can create a custom renaming format like:
{prefix}-{title}.pdf

Also, this field would be great if it is searchable.

Great application, I really like it!

**Comment by hany606** [comment]
Yes, I agree with you. I would prefer to modify this prefix field and automatically added to the name without the direct access via the OS file manager, because I think it corrupt the linking between the file and the item.

Also, when the file is renamed, it is easy to open the file directly using spotlight in mac

**Comment by GeoffreyChen777** [comment]
Ah, I got you, it's for spotlight.

Currently, it's unsupported.

But I would say I have a bigger plan for this.

What if I add three custom fields for the metadata.

```typescript
{
  title: string,
  author: string,
  ...
  customField1: {
    name: string,
    value: string
  },
  customField2: ...,
  customField3: ...
}
```

Users can rename the customField, input your prefered value via the editing view, and display it in the mainview.

Also you can use it for the file renaming.

Most importantly, It would be good for extension developers as they can display some custom information in the custom field.

**Comment by hany606** [comment]
That looks a good idea!
